domene/src/main/kotlin/no/nav/tiltakspenger/meldekort/service/OpprettMeldekortBehandlingService.kt

```diff
@@ -0,0 +1,46 @@
+package no.nav.tiltakspenger.meldekort.service
+
+import arrow.core.getOrElse
+import mu.KotlinLogging
+import no.nav.tiltakspenger.libs.common.CorrelationId
+import no.nav.tiltakspenger.libs.common.HendelseId
+import no.nav.tiltakspenger.libs.common.SakId
+import no.nav.tiltakspenger.libs.common.Saksbehandler
+import no.nav.tiltakspenger.libs.persistering.domene.SessionFactory
+import no.nav.tiltakspenger.meldekort.domene.opprettMeldekortBehandling
+import no.nav.tiltakspenger.meldekort.ports.MeldekortRepo
+import no.nav.tiltakspenger.saksbehandling.service.sak.SakService
+
+class OpprettMeldekortBehandlingService(
+    val sakService: SakService,
+    val meldekortRepo: MeldekortRepo,
+    val sessionFactory: SessionFactory,
+) {
+    private val logger = KotlinLogging.logger {}
+
+    suspend fun opprettBehandling(
+        hendelseId: HendelseId,
+        sakId: SakId,
+        saksbehandler: Saksbehandler,
+        correlationId: CorrelationId,
+    ) {
+        logger.info { "Oppretter behandling av $hendelseId på sak $sakId" }
+
+        val sak = sakService.hentForSakId(sakId, saksbehandler, correlationId).getOrElse {
+            throw IllegalArgumentException("Kunne ikke hente sak med id $sakId")
+        }.also {
+            if (it.hentMeldekortBehandling(hendelseId) != null) {
+                throw IllegalStateException("Det finnes allerede en behandling av $hendelseId: ${it.id}")
```

Comment:
Vi bør enten legge inn sperre på dette i frontend, informere alle brukerne våre godt (inkludert oss selv), eller propagere feilmeldingen til frontend. Hvis ikke gjetter jeg på dette vil støye litt

Reply:
Jepp, tenker ikke at funksjonen som å opprette behandlinger skal være tilgjengelig på meldeperiode-instanser som allerede har en behandling. Denne feilen skal bare kunne oppstå f.eks. hvis to saksbehandlere prøver å opprette på samme meldeperiode.